cmake/CheckAtomic.cmake

```diff
@@ -0,0 +1,56 @@
+# atomic builtins are required for threading support.
```

Comment:
This file needs to have ownership and license statement added.  (Copyright boilerplate.)  Where id this come from?

Reply:
It's an edit of the CheckAtomic Cmake module in LLVM. That version checked for CXX atomics, so I simplified it and wrote cases that test straight C atomics. There wasn't any owner information in the file or its context (it was in /usr/lib/llvm-7/lib/cmake/llvm/CheckAtomic.cmake rather than the source tree). I can try to track that down tomorrow.

Reply:
I'm happy to assign copyright of my changes to staysail if that makes things easier.

Reply:
No need to assign copyright -- but I want to make sure we give credit and observe the license of the original submitter.  I generally would prefer that this one be liberally licensed as well -- so if its some copy left then I'll probably decline and write my own checks.

Reply:
I wrote the test code itself based on tests I'd done myself for autoconf. All I got from LLVM was the cmake template since it was my first module. I've added the ownership/license statement (MIT license).

Reply:
Thanks :-)